src/eth1spec/spec.py

```diff
@@ -75,21 +78,30 @@ def state_transition(chain: BlockChain, block: Block) -> None:
     assert trie.TRIE(trie.y(state)) == block.header.state_root
 
 
-def verify_header(header: Header) -> bool:
+def verify_header(header: Header, parent_header: Header) -> bool:
     """
     Verifies a block header.
 
     Parameters
     ----------
     header : `eth1spec.eth_types.Header`
         Header to check for correctness.
+    parent_header: `eth1spec.eth_types.Header`
+        Parent Header of the header to check for correctness
 
     Returns
     -------
     verified : `bool`
         True if the header is correct, False otherwise.
     """
-    raise NotImplementedError()  # TODO
+    assert header.difficulty == calculate_block_difficulty(
+        header, parent_header
+    )
+    assert validate_gas_limit(header.gas_limit, parent_header.gas_limit)
```

Comment:
> This comment extends beyond this PR and thus shouldn't block the merge of this PR.

Whatever is chosen, we should be really consistent about this.  In general, I've found validation functions that return booleans to be foot-guns, but that stems from working for a long time under the convention that validation functions should raise exceptions or return `None`.  This is of course heavily influenced by writing "library" code, which this is not....

boolean return pattern is pretty easy to understand, and this model of many lines of `assert, assert, assert` is visually consistent.  I think it would be worth documenting some of the "style/convention" choices so that there is consistency in how the various pieces of this code get written.

Reply:
110% agree we need to document our coding conventions. Opened #42 to track.

Reply:
The maximum amount of agreement is 100%.  I call malarkey, shenanigans, or potentially some form of witchcraft.